src/Fixer/ControlStructure/NoUselessElseFixer.php

```diff
@@ -100,21 +100,24 @@ private function fixElse(Tokens $tokens, $index)
                 $previous,
                 -1,
                 array(
-                    ';',
-                    array(T_CLOSE_TAG),
+                    // positives
                     array(T_BREAK),
                     array(T_CONTINUE),
                     array(T_EXIT),
                     array(T_GOTO),
                     array(T_RETURN),
                     array(T_THROW),
+                    // negatives
+                    ';',
+                    array(T_IF),
```

Comment:
What would be even better/needed is to have a limit for the `getTokenOfKindSibling`method.